src/toil/server/cli/wes_cwl_runner.py

```diff
@@ -0,0 +1,342 @@
+import argparse
+import json
+import logging
+import os
+import subprocess
+import sys
+import time
+from io import BytesIO
+from urllib.parse import urlparse, urldefrag
+
+import requests
+import ruamel.yaml
+from typing import Optional, Dict, Any, List, Tuple, Iterable, cast
+from toil.wdl.utils import get_version as get_wdl_version
+
+from werkzeug.utils import secure_filename
+from wes_client.util import WESClient, wes_reponse as wes_response  # type: ignore
+
+
+"""
+A CWL runner that submits a workflow to a WES server, waits for it to finish,
+and outputs the results.
+
+Example usage with cwltest:
+
+```
+cwltest --verbose \
+    --tool=toil-wes-cwl-runner \
+    --test=src/toil/test/cwl/spec_v12/conformance_tests.yaml \
+    -n=0-50 \
+    --timeout=2400 \
+    -j2 \
+    -- \
+    --wes_endpoint=http://localhost:8080 \
+    --disableCaching \
+    --clean=always \
+    --logDebug
+```
+"""
+
+logger = logging.getLogger(__name__)
+
+
+class WESClientWithWorkflowEngineParameters(WESClient):  # type: ignore
+    """
+    A modified version of the WESClient from the wes-service package that
+    includes workflow_engine_parameters support.
+
+    TODO: Propose a PR in wes-service to include workflow_engine_params.
+    """
+    def __init__(self, endpoint: str, auth: Optional[Tuple[str, str]] = None) -> None:
+        """
+        :param endpoint: The http(s) URL of the WES server. Must include the
+                         protocol.
+        :param auth: Authentication information that will be attached to every
+                     request to the WES server.
+        """
+        proto, host = endpoint.split("://")
+        super().__init__({
+            "auth": auth,
+            "proto": proto,
+            "host": host
+        })
+
+    @staticmethod
+    def get_version(extension: str, workflow_file: str) -> str:
+        """Determines the version of a .py, .wdl, or .cwl file."""
+        if extension == "py":
+            return "3.8"
```

Comment:
This is mostly a placeholder for now. If we ever want it to be accurate, we could ask the user to provide it at least for Toil workflows.

Reply:
I think that should be recorded in a comment here. We want whoever looks at the code next not to waste a lot of time when they see this and realize it can't possibly actually work.
```suggestion
            # For Toil testing, we only ever run CWL workflows.
            # Just pretend all Python workflows are version 3.8 for now.
            return "3.8"
```